barretenberg/cpp/src/barretenberg/vm2/constraining/relations/instr_fetching.test.cpp

```diff
@@ -0,0 +1,146 @@
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <cstdint>
+#include <memory>
+#include <vector>
+
+#include "barretenberg/vm2/constraining/testing/check_relation.hpp"
+#include "barretenberg/vm2/generated/columns.hpp"
+#include "barretenberg/vm2/generated/flavor_settings.hpp"
+#include "barretenberg/vm2/generated/relations/instr_fetching.hpp"
+#include "barretenberg/vm2/testing/macros.hpp"
+#include "barretenberg/vm2/tracegen/bytecode_trace.hpp"
+#include "barretenberg/vm2/tracegen/precomputed_trace.hpp"
+#include "barretenberg/vm2/tracegen/test_trace_container.hpp"
+
+namespace bb::avm2::constraining {
+namespace {
+
+using tracegen::BytecodeTraceBuilder;
+using tracegen::TestTraceContainer;
+using FF = AvmFlavorSettings::FF;
+using C = Column;
+using instr_fetching = bb::avm2::instr_fetching<FF>;
+using simulation::Operand;
+
+TEST(InstrFetchingConstrainingTest, EmptyRow)
+{
+    TestTraceContainer trace({
+        { { C::precomputed_clk, 1 } },
+    });
+
+    check_relation<instr_fetching>(trace);
+}
+
+// Basic positive test with a hardcoded bytecode for ADD_8
+TEST(InstrFetchingConstrainingTest, add8WithTraceGen)
+{
+    TestTraceContainer trace;
+    BytecodeTraceBuilder builder;
+    simulation::Instruction add_8_instruction = {
```

Comment:
since you'll be using it a lot, and this is a test, do `using simulation::Instruction` at the top